### PR TITLE
Allow to disable caching in PooledByteBufAllocator for non FastThread…

### PR DESCRIPTION
**Pull request by normanmaurer**
…LocalThreads

Motivation:

If a user allocates a lot from outside the EventLoop we may end up creating a lot of caches in the PooledByteBufAllocator. This may be wasteful and so it may be useful for an other to configure that caches should only be used from within EventLoops.

Modifications:

Add new constructor which allows to configure the caching behaviour.

Result:

More flexible configuration of PooledByteBufAllocator possible

### CODE REVIEW

**Comment by normanmaurer** [comment]
@liuhualh FYI as well

File: buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java [hunk 1]
```diff
@@ -141,23 +147,42 @@ public PooledByteBufAllocator() {
...
+     * {@link PooledByteBufAllocator#PooledByteBufAllocator(boolean, int, int, int, int, int, int, int, boolean)}
+     */
+    @Deprecated
     public PooledByteBufAllocator(boolean preferDirect, int nHeapArena, int nDirectArena, int pageSize, int maxOrder,
                                   int tinyCacheSize, int smallCacheSize, int normalCacheSize) {
+        this(preferDirect, nHeapArena, nDirectArena, pageSize, maxOrder, tinyCacheSize, smallCacheSize,
+                normalCacheSize, DEFAULT_USE_CACHE_FOR_ALL_THREADS);
+    }
+
+    public PooledByteBufAllocator(boolean preferDirect, int nHeapArena, int nDirectArena, int pageSize, int maxOrder,
```
Comment (netkins) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: MAJOR Constructor has 9 parameters, which is greater than 7 authorized. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS00107)

Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java]: its all good :)

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (2b8fd8d43be916e29836943fc9d57ec494251756) and 4.0 (c131ef9f6f0f2effdda9d98f87b3bd05375d7002)

**Comment by liuhualh** [comment]
@normanmaurer from call stack on #5930 , the cache can not be GC is allocated from SupportedCipherSuiteFilter.filterCipherSuites,  is this feature have influence on this allocation path ?

**Comment by normanmaurer** [comment]
@liuhualh this issue was fixed by 705e3f629ab802539586c7d92f7888697a44c450

**Comment by liuhualh** [comment]
I see, Thank you very much

**Comment by normanmaurer** [comment]
@liuhualh np... I will release a new netty version this week which will include it.

**Pull request closed without merging**